Sources/ImageFetcher/Networking.swift

```diff
@@ -30,21 +30,54 @@ import Foundation
 /// A simple wrapper for a closure performing an async network request.
 public struct Networking {
     /// Downloads the contents of a URL based on the specified URL request and delivers the data asynchronously.
-    public let load: (URLRequest) async throws -> (Data, URLResponse)
+    public let load: (URLRequest) async throws -> Data
+
+    /// Creates a wrapper to perform async network requests.
+    /// - Parameter load: A closure to load a request.
+    public init(load: @escaping (URLRequest) async throws -> Data) {
+        self.load = load
+    }
 }
 
 public extension Networking {
+
+    /// A type that validates network responses.
+    enum ResponseValidator {
+        /// Throws an error if the given response was not successful.
+        /// - Parameter response: The response to validate.
+        public static func validate(_ response: URLResponse) throws {
```

Comment:
This doesn't feel quite right but

- it feels too specialized for a free function
- it doesn't feel right as a helper method
- I wanted to make it public so that it could be reused as part of a more specialized implementation of of `Networking.load`.

Another option would be 

```swift
struct ResponseValidator {
    let validate: (URLResponse) throws -> Void

   static var live: Self { ... }
}
```

but I didn't see any benefit to the added complexity, however small it might be.